src/convert.cpp

```diff
@@ -118,41 +115,29 @@ ddwaf_object* to_ddwaf_object(
   }
   if (val.IsNumber()) {
     mlog("creating Number");
-    if (coerceToString) {
-      return ddwaf_object_string(object, val.ToString().Utf8Value().c_str());
-    } else {
-      return ddwaf_object_string_from_signed(object, val.ToNumber().Int64Value());
-    }
+    return ddwaf_object_string_from_signed(object, val.ToNumber().Int64Value());
   }
   if (val.IsBoolean()) {
     mlog("creating Boolean");
     bool boolValue = val.ToBoolean().Value();
-    if (coerceToString) {
-      return ddwaf_object_string(object, boolValue ? "true" : "false");
-    } else {
-      return ddwaf_object_bool(object, boolValue);
-    }
+    return ddwaf_object_bool(object, boolValue);
   }
   // TODO(carles): BigInt is not available for NodeJs <14. Enable it when dropping support for NodeJs 12
   /*
   if (val.IsBigInt()) {
     mlog("creating BigInt");
     bool lossless;
     int64_t intValue = val.As<Napi::BigInt>().Int64Value(&lossless);
-    if (coerceToString || !lossless) {
-      return to_ddwaf_string(object, val, lim);
-    } else {
-      return ddwaf_object_string_from_signed(object, intValue);
-    }
+    return ddwaf_object_string_from_signed(object, intValue);
```

Comment:
```suggestion
    if (!lossless) {
      return to_ddwaf_string(object, val, lim);
    } else {
      return ddwaf_object_string_from_signed(object, intValue);
    }
```

Reply:
I'm going to move this code to a Jira ticket so it's tackled when we drop node 12 from the build process. I find a little absurd maintaining a commented code block.

Reply:
That's fair haha

Reply:
You could also just create an issue with it :+1: